app/assets/javascripts/discourse/app/controllers/user-activity-bookmarks.js

```diff
@@ -30,11 +30,11 @@ export default class UserActivityBookmarksController extends Controller {
 
   @computed("q")
   get searchTerm() {
-    return this.q;
+    return this._searchTerm || this.q;
```

Comment:
This likely means you can't clear the filter. Might leave it as it was, it wouldn't be perfect, it would mean that when going to `?q=test` and hitting search, the results would reset. But that should be a very very rare use case.